File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_09/java/CQueue.java
package com.hi.dhl.algorithms.offer._09.java;

import java.util.Stack;

/**
 * <pre>
 *     author: dhl
 *     desc  : 两个栈实现一个队列
 * </pre>
 */

class CQueue {
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public CQueue() {
        stack1 = new Stack();
        stack2 = new Stack();
    }

    public void appendTail(int value) {
        stack1.push(value);
    }

    public int deleteHead() {
        if (!stack2.isEmpty()) {
            return stack2.pop();
        } else if (stack1.isEmpty()) {
            return -1;
        } else {
            while (!stack1.isEmpty()) {
                stack2.push(stack1.pop());
            }
            return stack2.pop();
        }
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_09/kotlin/CQueue.kt
package com.hi.dhl.algorithms.offer._09.kotlin

import java.util.*

/**
 * <pre>
 *     author: dhl
 *     desc  : 两个栈实现一个队列
 * </pre>
 */

class CQueue() {
    val stack1 = Stack<Int>()
    val stack2 = Stack<Int>()

    fun appendTail(value: Int) {
        stack1.push(value);
    }

    fun deleteHead(): Int {
        return when {
            !stack2.isEmpty() -> stack2.pop()
            stack1.isEmpty() -> -1
            else -> {
                while (!stack1.isEmpty()) {
                    stack2.push(stack1.pop())
                }
                return stack2.pop()
            }
        }
    }
}